### DISCUSSION TITLE
Client type breaks with "collides with built-in method" across packages in mono repo

### DISCUSSION THREAD

**Discussion by jimkyndemeyer** [trpc/trpc#3780] [Q&A] [answered]
Hey,

I've defined a trpc client in a package inside a mono repo. Inside the package the type of the client according to TypeScript is resolved to:

```typescript
export const client: DecoratedProcedureRecord<{ping: BuildProcedure<"query", {_config: RootConfig<{ctx: object, meta: object, errorShape: DefaultErrorShape, transformer: DefaultDataTransformer}>, ... 5 more ..., _meta: object}, string>, greeting: BuildProcedure<...>, anotherQuery: BuildProcedure<...>}, CreateRouterInner<...>>
```

In that package I get type-safe access to the different queries and mutations on the client, e.g. `greeting` from the example.

But when I try to use the client from another package in the mono repo the client type is resolved to the following by TypeScript:

```typescript
export const client: "query collides with a built-in method, you should rename this router or procedure on your backend" | "mutation collides with a built-in method, you should rename this router or procedure on your backend" | ... 5 more ... | "requestAsPromise collides with a built-in method, you should rename this router or procedure on your backend"
```
The client does work, so this is a type only problem, and I haven't defined a query named `query` or a mutation called `mutation`.

Any ideas how to resolve this?

I've tried trpc versions 10.9.0 - latest. I'm using TypeScript 4.9.3.

**Update:**

With version 10.9.0 the type of the client becomes somewhat valid instead of the error string type, but queries are coming back as `any` which means there's no type safety on inputs or return values. Hmm...

**Comment by jimkyndemeyer** [comment, marked as answer]
**If someone else encounters this issue:**

The type error was due to using `tsconfig.json` `"paths"` in the mono repo package that declared the client.

When used from other packages in the mono repo, each with their own `tsconfig,json`, the use of said paths inside the trpc package (e.g. `import type {AppRouter} from '~/pages/api/trpc/[trpc]'`) breaks the exported type of the client.

With relative import paths in the trpc package, the client type works across mono repo packages as expected.

**Comment by emirefek** [comment]
adding composite:true to servers workspace's tsconfig file fixed for me. Also added paths and refferances
